wasm/instructions.cpp

```diff
@@ -171,4 +176,71 @@ std::optional<std::vector<Instruction>> parse(std::istream &is) {
     }
 }
 
+std::optional<std::string> to_string(Instruction const &inst) {
+    std::string out;
+
+    if (std::holds_alternative<Block>(inst)) {
```

Comment:
If we replace this `if-else if`-chain w/ `std::visit`, we can get a compilation error if any of the variants in `Instruction` are unhandled, and that would allow us to always return a `std::string` instead of an optional one.